app/src/main/java/com/lms/mpasho_lms_news/view/MainActivity.java

```diff
@@ -122,4 +146,49 @@ public boolean onQueryTextChange(String newText) {
 
         return true;
     }
+
+    @Override
+    public void onRefresh() {
+        LoadJson("");
+    }
+
+    private void onLoadingSwipeRefresh(final String keyword) {
+        swipeRefreshLayout.post(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        LoadJson(keyword);
+                    }
+                }
+        );
+    }
+
+    private void initListener() {
+
+        adapter.setOnItemClickListener(new Adapter.OnItemClickListener() {
+            @Override
+            public void onItemClick(View view, int position) {
+                ImageView imageView = view.findViewById(R.id.img);
+                Intent intent = new Intent(MainActivity.this, NewsDetailActivity.class);
+
+                Article article = articles.get(position);
```

Comment:
is there a better way of accessing this data at the destination?
Or you can pass the full object and not the getters and setters then retrieve them from the destination class.